### ISSUE TITLE
REQ_WCAG_1.2.1 Audio-only

### ISSUE THREAD

**Issue by ecureuill** [solex2006/SELIProject#8]
# GUIDELINE
## Principle
> PERCEIVABLE: Information and user interface components must be presentable to users in ways they can perceive. This means that users must be able to perceive the information being presented (it can't be invisible to all of their senses) Provide text alternatives for non-text content. Provide captions and other alternatives for multimedia. Create content that can be presented in different ways, including by assistive technologies, without losing meaning. Make it easier for users to see and hear content.

## Level of Conformance 
> **A** (the minimum level of conformance), the Web page satisfies all the Level A Success Criteria, or a conforming alternate version is provided

## Referenced Requirements
Reference requirements that should be meet together 
#9 

## General description
> An alternative for time-based media is provided that presents equivalent information for prerecorded audio-only content.

## Intent
> Alternatives for time-based media that are text based make information accessible because text can be rendered through any sensory modality (for example, visual, auditory or tactile) to match the needs of the user. In the future, text could also be translated into symbols, sign language or simpler forms of the language (future).

## Examples

1. An audio recording of a speech
The link to an audio clip says, "Chairman's speech to the assembly." A link to a text transcript is provided immediately after the link to the audio clip.
2. An audio recording of a press conference
A Web page includes a link to an audio recording of a press conference that identifies the audio recording. The page also links to a text transcript of the press conference. The transcript includes a verbatim record of everything the speakers say. It identifies who is speaking as well as noting other significant sounds that are part of the recording, such as applause, laughter, questions from the audience, and so on.

## Techniques

- [G158: Providing an alternative for time-based media for audio-only content](https://www.w3.org/WAI/WCAG21/Techniques/general/G158)
>This technique involves creating a document that tells the same story and presents the same information as the prerecorded audio-only content. In this technique, the document serves as long description for the content and includes all of the important dialogue and as well as descriptions of background sounds etc. that are part of the story.

### Procedure

1. View the audio-only content while referring to the alternative for time-based media.
2. Check that the dialogue in the transcript matches the dialogue and information presented in the audio-only presentation.
3. If the audio includes multiple voices, check that the transcript identifies who is speaking for all dialogue.
4. Check that at least **one of the following is true**:

- The transcript itself can be programmatically determined from the text alternative for the audio-only content
- The transcript is referred to from the programmatically determined text alternative for the audio-only content

5. If the alternate version(s) are on a separate page, check for the availability of link(s) to allow the user to get to the other versions.

### Expected Results

All of the above checks are true.

# SELI CONTEXT
> How to implement this requirement on SELI Platform?

```
<div id="id_video_box">
   <h3>Audio title</h3>
   <p id="id_short_desc"><!-- SHORT DESCRIPTION --></p>
   <h3><button aria-expanded="false" aria-controls="id_req_acc">Available accessibility resources</button></h3>
   <ul id="id_req_acc">
  	<li>Transcriptions</li>
   </ul>
   <audio id=audio autoplay="false" autoplay="controls" aria-describedby="id_long_desc" aria-labelby="id_short_desc">
       <source src="audio.mp3" type='audio/mpeg'/>
   </audio>
</div>
<button aria-expanded="false" aria-controls="id_long_desc">Audio Transcription</button>
<div><p  id="id_long_desc"><!-- TRANSCRIPTIONS --></p></div>
```
## Tasks
List the tasks required to accomplish the requirement (not exhaustive)

- [ ] **DEVELOPER** MUST include audio-only contents with it text alternative
> All contents not created by users (e.g. SELI "about" section text, buttons, fields) that STUDENT could consume 

- [ ] When **TEACHER** upload an audio-only content **SYSTEM** MUST provide a field to **TEACHER** write/paste a text transcription 

- [ ] **SYSTEM** MUST render teacher's content with it respective text alternative, if teacher configured one

# :notebook_with_decorative_cover:   References

- [How to Meet WCAG (Quick Reference)](https://www.w3.org/WAI/WCAG21/quickref/#audio-only-and-video-only-prerecorded)
- [Understanding Techniques for WCAG Success Criteria](https://www.w3.org/WAI/WCAG21/Understanding/understanding-techniques) 
- [aria-describedby for longer text alternative](https://www.w3.org/WAI/PF/HTML/wiki/Media_Alt_Technologies#2:_Use_.40aria-describedby_for_the_longer_text_alternative)
- [Transcript Examples](https://www.opaltranscriptionservices.com/transcript-example/)

# GLOSSARY 
:warning: The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119/8174](http://www.rfc-editor.org/info/rfc8174).
> 1. **MUST** This word, or the terms "REQUIRED" or "SHALL", mean that the definition is an absolute requirement of the specification.
> 2. **MUST NOT** This phrase, or the phrase "SHALL NOT", mean that the definition is an absolute prohibition of the specification.
> 3. **SHOULD** This word, or the adjective "RECOMMENDED", mean that there may exist valid reasons in particular circumstances to ignore a particular item, but the full implications must be understood and carefully weighed before choosing a different course.
> 4. **SHOULD NOT** This phrase, or the phrase "NOT RECOMMENDED" mean that there may exist valid reasons in particular circumstances when the particular behavior is acceptable or even useful, but the full implications should be understood and the case carefully weighed before implementing any behavior described with this label.
> 5. **MAY**   This word, or the adjective "OPTIONAL", mean that an item is truly optional. An implementation which does not include a particular option MUST be prepared to interoperate with another implementation which does include the option, though perhaps with reduced functionality. In the same vein an implementation which does include a particular option MUST be prepared to interoperate with another implementation which does not include the option (except, of course, for the feature the option provides.)

**Comment by ecureuill** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [REQ_WCAG_1.2.1 Audio-only](https://trello.com/c/5fgcII7b/1-reqwcag121-audio-only)
